### ISSUE TITLE
"Equation" accordion box should be a sbclass of AccordionBox.

### ISSUE THREAD

**Issue by pixelzoom** [phetsims/curve-fitting#173]
Related to https://github.com/phetsims/sun/issues/919#issuecomment-2546357326 ...

EquationAccordionBox (screenshot below) should be a subclass of AccordionBox, not a subclass of Panel with its own ExpandCollapseButton. This is important for PhET-iO and a11y features. 

<img width="297" alt="screenshot_3652" src="https://github.com/user-attachments/assets/ba80cbfc-e25c-4800-9215-e1e854067593" />

This issue is deferred until work resumes on this sim.
